docs/md/expressions.md

```diff
@@ -393,15 +393,46 @@ It is the inverse of [char](expressions.html#function:char).
 @(code("")) → ERROR
 ```
 
-<a name="function:datetime"></a>
+<a name="function:date"></a>
 
-## datetime(text)
+## date(value)
 
-Parses `text` into a date using to the default date format.
+Tries to convert `value` to a date.
 
+If it is text then it will be parsed into a date using to the default date format.
```

Comment:
"if it is text" may be a bit confusing to people.. its all text to them :)

Reply:
I think (at least some of our) users understand that when they do `1 + 2` or `x != "bar"` they are working with non-text types

Reply:
`1+2` sure, but I think its a lot to expect for them to reason about something that is stringy looking not being a string. Whatever, keep it if you want, but I think we should be careful with this kind of language and remember their point of view.

Reply:
I would just rather see something like "if it is not an ISO8601 date" which would provide a bit more clue as to how this is going to behave.

Reply:
But that's not even accurate - we'll parse lots of things that aren't ISO8601 dates

Reply:
isn't saying "If this isn't an ISO 8601 date then this will be parsed according to your org settings" accurate?

Reply:
No because `DATE(2019)` won't try to parse anything. It'll say sorry we can't convert a number to a date.

Reply:
But `DATE("2019")` does work? Does that feel right to you?

Reply:
`DATE("2019")` will return an error saying `"2019"` couldn't be parsed as a date... yes that feels right

Reply:
I feel like that is way too subtle. Remember the parameter may be a contact field or a flow value that is an integer and to the user that is going to be really weird that those behave differently. When they echo that value it is going to look identical. `2019` and `2019` Why wouldn't we try to cast the argument to a string so we are consistent in both?

Reply:
I honestly don't get why it's a bad thing to give the user some insight into what types are being created by different expressions, and supposed I disagree with your fundamental premise that users are oblivious to intermediate types :):):):)

`DATE(2019+2+19)` returning an error saying that a number couldn't be converted to a `DATE` gives them insight into what `2019+2+19` does. 

If they're using a contact field it might remind them that fields can be `TEXT` or `NUMBER` etc

Reply:
You keep using literals, but that's not going to be the typical case.

Getting that error for `DATE(contact.birth_year)` is going to be super weird to a user, especially when they echo `@contact.birth_year` and `@contact.birth_year_string` and they look the same.

We are creating shit work for a user to make them cast into a string, why would we do that when we can try it ourselves. I don't understand the argument against it, can you explain that, I must be missing something.

Reply:
If `contact.fields.birth_year` is a NUMBER like `1980`.. then I think the ideal error message for `DATE(contact.fields.birth_year)` is 

`the number 1980 couldn't be converted to a date`

It reminds the user that `contact.fields.birth_year` is a number field.

"We are creating shit work for a user to make them cast into a string"

I'm not following.. what to do they need to cast to a string?

Reply:
(not trying to kill this conversion but I need to merge this to fix a ticket...)

Reply:
Ok, maybe this isn't such a big deal for `DATE` since it doesn't accept a single year either way, but what happens in the `TIME` case where we do? Do we accept both `"11"` and `11`? (or any other function for that matter)

My concern is places where we act one way vs another for a type that the user can't really see. I can be convinced different errors are maybe ok though I don't love it, but I'm not ok with one succeeding for `"10"` and failing for `10` or vice versa.

Reply:
We don't currently accept `TIME(10)` but that's only cuz I forgot to update that when we decided a single hour was a valid time.

I think that's actually a useful principle moving forward - that `foo(text(bar))` should work if `foo(bar)` works. I just think it's beneficial to let the user see the actual type of thing passed to `foo` if it fails (and certainly more beneficial to one of us trying to debug a flow).

Reply:
Ok, then we are in agreement. I think the inverse should also be true `foo(bar)` should succeed if `foo(int(bar))` does.